### PR TITLE
Code cleanup and multiple currencies

### PR DESCRIPTION
**Pull request by Wailord**
UI can still use a lot of work but I'll continue working on that; wanted to get the ball rolling on this. I think there was a lot of separation of concerns violations that I tried to tidy up a bit, which should also allow for a lot more robust functionality going forward with fewer bugs. I switched APIs to use CMC as it is far more useful when supporting multiple currencies; CMC + the way networking is now handled means adding new currencies is as simple as adding a case in an enum and the ticker that represents it. If we're cool with this moving forward I'm working with a designer and we can spiffy up the UI next :)

To enumerate what changed from an end-user perspective:
- Change between BTC/USD/EUR for price by tapping currency name
- Add text to clarify what the price is
- Tap away the numeric keyboard
- A lot more validation for the numeric keyboard
- Use CMC over BG/Mercatox

Most of the changes were code infrastructure that I'd be happy to discuss.

### CODE REVIEW

File: Rai/Network/PriceController.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//
+//  PriceController.swift
+//  Rai
+//
+//  Created by Ryan Fox on 12/22/17.
+//  Copyright © 2017 Zack Shapiro. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
```
Comment (zackshapiro) [inline Rai/Network/PriceController.swift:10]: First, thanks for the PR!

I don't want to use SwiftyJSON, and should remove it from the Podfile. Swift 4's `Decodable` protocol is a much nicer way to encode/decode.

CMC's API has a rate limit and that's why I was avoiding it. Maybe write a fallback from Merc's API call to hit CMC if both initial APIs are offline?

**Comment by zackshapiro** [comment]
Can you provide screenshots?

**Comment by zackshapiro** [comment]
After looking more into the code, I'm not interested in merging these changes other than the USD/EUR implementation, maybe the keyboard handling and better validation in the text field. If you want to submit just those changes in a separate PR, I'd be happy to take a look.

A few reasons behind my thinking:
1) Seems like you re-wrote most of the app in a style that you like, some I'm sure for valid reasons, some because our styles differ. This is also a small side project of mine and I don't think a re-write is needed. Some things could definitely be shared among the main code and the today widget but nothing drastic.
2) There's a combination of layout styles, one using cartography and one using NSLayoutConstraints that just adds complexity
3) The Decodable/SwiftyJSON note from above
4) I don't care about academics 😄 I mean that in the nicest possible way. I care about running code and a working product and not the way something "should" be implemented for a project like this.

Thanks!

File: Rai/Network/PriceController.swift (hunk 1 above)
Comment (Wailord) [reply Rai/Network/PriceController.swift:10]: Yeah, I wasn't quite sure if you were moving towards SwiftJSON or away from it. Agreed that adopting as much as Swift gives us as possible is best!

As for CMC, their rate limits seem pretty generous -- one every six seconds. Would you expect to exceed that?

**Comment by Wailord** [comment]
1) It was _mostly_ changes necessitated by what I ended up adding for currency conversion; because adding currency conversion meant a new button, a new picker view, and more view handling, the code just kept growing ViewController.swift.
2) Agreed that this is an issue; I haven't used Cartography before, so I ended up writing without thinking of how to convert it.
3) Agreed.
4) Yeah, I don't want to give off the impression that I'm over-engineering for the sake of over-engineering. As the requirements grew so too did the importance of not muddling the things I was adding (logic around adding the picker, handling selection, firing off a request as a result, updating the UI when it came back). It was getting too spaghetti-heavy as I kept writing. Do you think the right approach is to keep that logic inside of ViewController.swift?

Thanks for taking the time to review!

**Pull request closed without merging**